### ISSUE TITLE
xmlhttprequest not working for cross-domain with IE8

### ISSUE THREAD

**Issue by caibaohua** [metajack/strophejs#49]
Hi, 
I found under IE8, the strophejs cannot connect the server when cross-domain. And there is a plugin to fix it by using xDomainRequest which still not working.
Is there other solutions for fix this problem?

Thanks,

Baohua

**Comment by CharlieDigital** [comment]
Strophe itself has examples using flensed (Flash based) for cross domain HTTP requests.  You should give that a try.

**Comment by salomvary** [comment]
You should also consider trying this:
https://github.com/salomvary/strophejs-crossxhr
(Although it only helps if only the _subdomains_ differ. Protocol and port must be the same.)

**Comment by radsaq** [comment]
If you're using XDomainRequest, make sure that your server is sending the right response headers to allow CORS.

**Comment by kenstir** [comment]
The strophe.xdomainrequest.js plugin works for me with a small edit:
https://gist.github.com/1095825/6b4517276f26b66b01fa97b0a78c01275fdc6ff2

And as radsaq says, make sure your server is sending the Access-Control-Allow-Origin header, and that you are living within the same-scheme limitations of the XDomainRequest object.

**Comment by yankov** [comment]
What is the current best solution to make strophe work crossdomain? Taking in account that need to cover as much browsers as possible: means no flash sockets.

**Comment by radsaq** [comment]
Sadly, it's probably to load either the flxhr, cors, or xdr plugins depending on the browser. If you're not supporting IE7 you could possibly get away without flxhr.

**Comment by kenstir** [comment]
The xdr plugin works well for me, but it does not work for ie7:

https://gist.github.com/1095825

## 

-Ken

**Comment by astro** [comment]
Would you like to see this plugin in upstream Strophe.js? Who is the author and does he agree publishing under Strophe's MIT license?

**Comment by kenstir** [comment]
Yes I would like to see it upstream, and you should feel free to ask the author: https://gist.github.com/iadvize

**Comment by astro** [comment]
Thanks, I just sent him a message.

I fear I don't have the resources to test in IE. Would you mind collecting all the changes in this Github issue for a pull request?

**Comment by astro** [comment]
Big news everyone: we can use it.

iadvize said:

> but take into account that since IE9 was released, this plugin can cause
> problem when compatibility mode is enabled
> i did not manage to find why.

I would appreciate a well-tested pull request.
